### PR TITLE
Knitby new animation

### PR DESCRIPTION
**Pull request by jentaruno**
## Description
What changes does this PR include? Is there anything that affects other features that people should be aware of?
- Add Knitby new animations by Silvan
- Add Knitby animation control
- On player death, death smoke and respawn at player position
- The 144 files changed are mostly sprite files so don't be scared to review this ily

## Before and After
Screenshots/videos of the changes:
https://github.com/user-attachments/assets/9947a3fa-b235-4a18-ab02-7a9dda5d66cc

## Checklist
- [x] This code builds and runs
- [x] All public classes and methods are documented
- [x] Hard-to-understand areas of the code are documented
- [x] Self-review done

### CODE REVIEW

File: Assets/Scripts/KnitbyController.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Updates Knitby's position to follow the player's path
+/// </summary>
+public class KnitbyController : MonoBehaviour
+{ 
+    [Header("References")]
+    [SerializeField] private LineRenderer lineRenderer;
```
Comment (AdenC123) [inline Assets/Scripts/KnitbyController.cs:11]: This is forcing an override in Knitby- we should get the LineRenderer directly from the player during Awake instead.

Also, future task for animating: the swing line should be animated to appear from Knitby and then attach to the player. When we make that change the LineRenderer also should no longer be controlled from the PlayerController.

File: Assets/Prefabs/Knitby.prefab [hunk 2]
```diff
@@ -1,46 +1,46 @@
...
   m_ObjectHideFlags: 0
   m_CorrespondingSourceObject: {fileID: 0}
   m_PrefabInstance: {fileID: 0}
   m_PrefabAsset: {fileID: 0}
-  m_GameObject: {fileID: 8090917524103177417}
+  m_GameObject: {fileID: 657823012751151630}
   serializedVersion: 2
-  m_LocalRotation: {x: 0, y: 0, z: 0, w: 1}
-  m_LocalPosition: {x: -2.41, y: 0.34, z: 0}
-  m_LocalScale: {x: 1, y: 1, z: 1}
```
Comment (AdenC123) [inline Assets/Prefabs/Knitby.prefab:32]: Knitby looks a little small to me- I tried increasing the scale to 1.2 and it looked more natural. If that also looks better to you we can change it

**Review by AdenC123 — CHANGES_REQUESTED** [review]
(can github get custom emojis so I can put a knitby emoji here)

Few small suggestions for code and sprite size. Also, the Knitby walking animation seems slow and choppy to me, mostly because Knitby slides along the ground while walking. I think speeding it up a bit would help, but if that doesn't work then the animation speed should be set dynamically based on the game object's speed along the ground.

File: Assets/Scripts/KnitbyController.cs (hunk 1 above)
Comment (jentaruno) [reply Assets/Scripts/KnitbyController.cs:11]: going to change it when we change line renderer

Comment (AdenC123) [reply Assets/Scripts/KnitbyController.cs]: changed

**Comment by AdenC123** [comment]
LGTM 🧶🧶🧶

**Review by AdenC123 — APPROVED** [review]
YUP

**Pull request merged**